azurelinuxagent/common/protocol/wire.py

```diff
@@ -158,14 +158,20 @@ def get_artifacts_profile(self):
         logger.verbose("Get In-VM Artifacts Profile")
         return self.client.get_artifacts_profile()
 
+    def download_ext_handler_pkg_through_host(self, uri, destination):
+        host = self.client.get_host_plugin()
+        uri, headers = host.get_artifact_request(uri, host.manifest_uri)
+        success = self.client.stream(uri, destination, headers=headers, use_proxy=False)
+        return success
+
     def download_ext_handler_pkg(self, uri, destination, headers=None, use_proxy=True):
-        success = self.client.stream(uri, destination, headers=headers, use_proxy=use_proxy)
+        direct_func = lambda: self.client.stream(uri, destination, headers=None, use_proxy=True)
```

Comment:
Let's add this comment to all the instances of those lambdas, and something similar to send_request_using_appropriate_channel

        # NOTE: the host_func may be called after refreshing the goal state, be careful if you capture
        # any goal state data in the lambda.

Reply:
Added